pkg/apis/core/v1/defaults.go

```diff
@@ -217,6 +200,17 @@ func SetDefaults_Pod(obj *v1.Pod) {
 				}
 			}
 		}
+		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) && utilfeature.DefaultFeatureGate.Enabled(features.SidecarContainers) {
+			isRestartableInitContainer := false
+			c := obj.Spec.InitContainers[i]
+			if c.RestartPolicy != nil && *c.RestartPolicy == v1.ContainerRestartPolicyAlways {
+				isRestartableInitContainer = true
+			}
+			if isRestartableInitContainer && (c.Resources.Requests != nil || c.Resources.Limits != nil) {
```

Comment:
I am not able to use `podutil.IsRestartableInitContainer` since that function is defined in `pkg/api/v1/pod/util.go` where we are importing `v1 "k8s.io/api/core/v1"` (current package) leading to import cycle error.